content/tutorials/mqtt-snake.textile

```diff
@@ -0,0 +1,539 @@
+
+  ```[sh]
+    pip install paho-mqtt
+  ```
+
+h2. Step 3 - Add your libraries
+
+blang[javascript].
+  With your NPM modules ready, it's time to create a file to contain your controller code. Create a file called @controller.js@ inside your @mqtt-snake@ folder. Inside this file, instance the MQTT.js and keypress libraries:
+
+  ```[javascript]
+    const mqtt = require('mqtt');
+    const keypress = require('keypress');
+  ```
+
+  "See this step in Github":https://github.com/ably/tutorials/commit/mqtt-snake-js-step3
+
+blang[go].
+  With your libraries installed, it's time to create a file to contain your controller code. Create a file called @controller.go@ inside your @mqtt-snake@ folder. Inside this file, reference the libraries we'll need:
+
+  ```[go]
+    package main
+
+    import (
+      'fmt'
+      MQTT 'github.com/eclipse/paho.mqtt.golang'
+      termbox 'github.com/nsf/termbox-go"
+    )
+  ```
+
+  "See this step in Github":https://github.com/ably/tutorials/commit/mqtt-snake-go-step3
+
+blang[python].
+  With your libraries ready, it's time to create a file to contain your controller code. Create a file called @controller.py@ inside your @mqtt-snake@ folder. Inside this file, instance the paho-mqtt and curses libraries:
+
+  ```[python]
+    import paho.mqtt.client as mqtt
+    import curses
+  ```
+
+  "See this step in Github":https://github.com/ably/tutorials/commit/mqtt-snake-python-step3
+
+h2. Step 4 - Connect to Ably through MQTT
+
+With the libraries now available, it's time to set up MQTT. When using MQTT with Ably, there are a few requirements with regards to your setup. You'll need to connect to 'mqtt.ably.io' on port 8883, which requires the use of SSL/TLS in your connection. If you have a device which cannot support SSL, you'll need to connect via port 1883 instead, but this will come with a number of "restrictions":https://support.ably.io/solution/articles/3000074906-using-the-mqtt-protocol-adapter. 
+
+You will need to set the "keep alive" time value to between 15 and 60 seconds, where 60 seconds will maximize the battery life, and 15 seconds will maximize responsiveness to network issues. For this tutorial you'll be setting it to 15.
+
+For authentication, you'll need to either use an API Key, or a token. For simplicity of the tutorial, you'll be using the API key you obtained in step 1. For this, you'll need to provide the first half of the API Key (before the colon) as the username, and the second half of the API Key (after the colon) as the password. For example, if your API key is @A12B3C.4D:5E6F7G8H@, your username will be @A12B3C.4D@ and your password will be @5E6F7G8H@.
+
+blang[javascript].
+  Add the following code below your current code in @controller.js@, replacing 'FIRST_HALF_OF_API_KEY' and 'SECOND_HALF_OF_API_KEY' with the appropriate parts of your own API Key:
```

Comment:
Should this not be `USERNAME_PART_OF_API_KEY` and `PASSWORD_PART_OF_API_KEY` as the user either understands how API keys are split because they read the part befroe, or they don't and first & second half makes no sense then anyway?